File: app/src/main/java/fileio/ex02.java
package fileio;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class ex02 {
    public static void main(String[] args) {
        Path file = Path.of("app", "build", "tmp", "week08_ex02.txt");

        try {
            Files.createDirectories(file.getParent());
            try (BufferedWriter writer = Files.newBufferedWriter(file)) {
                writer.write("line1");
                writer.newLine();
                writer.write("line2");
            }

            try (BufferedReader reader = Files.newBufferedReader(file)) {
                String line;
                while ((line = reader.readLine()) != null) {
                    System.out.println("read: " + line);
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: app-kotlin/src/main/kotlin/fileio/ex02.kt
package fileio

import java.nio.file.Files
import java.nio.file.Path

class ex02 {
    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val file = Path.of("app-kotlin", "build", "tmp", "week08_ex02.txt")

            try {
                Files.createDirectories(file.parent)
                Files.newBufferedWriter(file).use { writer ->
                    writer.write("line1")
                    writer.newLine()
                    writer.write("line2")
                }

                Files.newBufferedReader(file).useLines { lines ->
                    lines.forEach { println("read: $it") }
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }
}
